### DISCUSSION TITLE
TempFlowEstimator and TransformerTempFlowEstimator with the given data

### DISCUSSION THREAD

**Discussion by sege01** [zalandoresearch/pytorch-ts#77] [Q&A]
Hello,

It would be very helpful if you can provide your thoughts on the below issue.
I have a dataset that looks like below:- 

- id--> unique id

- column between id and Stat_Cat1 are target values
- Stat_Cat_1 contains static categorical features.

![image](https://user-images.githubusercontent.com/85530006/133674606-0f1aa999-1519-4a85-88b0-48fe4c164519.png)

The original dataset is huge with around 10-12k rows and different static categorical and dynamic real features
Each row here represents a time series.
Is it possible to do probabilistic forecasting on this dataset using TempflowEstimator and TransformerTempFlowEstimator?
Is the below a correct way of creating a training dataset for these models?
```
train_ds = ListDataset([
    {
         FieldName.TARGET: targets.values,
         FieldName.START: start,
         FieldName.FEAT_STATIC_CAT: fsc,
   } 
     for (start,fsc) in zip( date_list.values,
                                 stat_cat,)

], freq="W",one_dim_target=False)
```

where targets is a dataframe containing target values,
          date_list contains the start date for the time series,
and stat_cat is
```
cat1_ids = df["Stat_Cat1"].astype('category').cat.codes.values
cat1_ids_un , cat1_ids _counts = np.unique(cat1_ids , return_counts=True)
cat2_ids = df["Stat_Cat2"].astype('category').cat.codes.values 
cat2_ids_un , cat2_ids _counts = np.unique(cat2_ids , return_counts=True)
uni_ids = df["id"].astype('category').cat.codes.values
uni_ids_un , uni_ids _counts = np.unique(uni_ids , return_counts=True)
stat_cat_list = [cat1_ids, cat2_ids]
stat_cat = np.concatenate(stat_cat_list)
stat_cat = stat_cat.reshape(len(stat_cat_list), len(uni_ids)).T
```

@kashif Could you please help by providing your thoughts on this?
It would be of great help.

Thank you so much.

**Comment by kashif** [comment]
@sege01 ok sure let me have a look at the issue and report back...

**Reply by sege01** [reply]
@kashif to simplify the issue:
Splitting the dataset into different multivariate groups according to the stat cat features and then passing it to the TempFlowEstimator and TransformerTempFlowEstimator? 

One example could be the m5 forecasting:
https://github.com/awslabs/gluon-ts/blob/master/examples/m5_gluonts_template.ipynb

What is the correct way to do it according to you if it is possible with the current model?

Thanks
